### ISSUE TITLE
macOS: initial opening via `open -a "Neovide"` does not set arglist

### ISSUE THREAD

**Issue by chrisgrieser** [neovide/neovide#2629]
**Describe the bug**
When opening Neovide via `open -a "Neovide" {filename}`, the arglist (as well as related values such as `argv` or `argc`) are not set. 

This means that using `open -a "Neovide"`, there is no way to tell if Neovide was opened with an argument or not. This is relevant for configs, where you want to open a dashboard on startup only when opening Neovide without argument.

**To Reproduce**
1. `open -a "Neovide" {filename}`
2. run `:lua = vim.fn.argc()`. Result is `0`
3. in comparison, `neovide {filename}` results in an `argc` of `1`

**Expected behavior**
arglist should not be empty.

**Screenshots**

**Desktop (please complete the following information):**
```
NVIM v0.10.0
Build type: Release
LuaJIT 2.1.1716656478
macOS 14.4.1 (arm)
neovide 0.13.1
```
[neovide_rCURRENT.log](https://github.com/user-attachments/files/15535559/neovide_rCURRENT.log)

**Comment by fredizzimo** [comment]
I'm not a macOS user, but I think 
`open -a Neovide --args {filename}` should do it. That will pass the arguments to Neovide instead of going through the finder. Or even `open -a Neovide --args -- {filename}`, which passes it directly to neovim.

I'm not sure if it can be fixed for the finder, since it uses the same functionality for drag and drop https://github.com/neovide/neovide/pull/2395/files.

**Comment by fredizzimo** [comment]
When looking at a different issue, I saw this https://developer.apple.com/documentation/appkit/nsapplicationdelegate/1428385-applicationdidfinishlaunching.

So, we should be able to wait for that until the list of files is passed to neovim. Or perhaps even delay the launching of neovim until that point, so that we can properly pass the files as a command line.

* See https://github.com/neovide/neovide/issues/2609, as it's probably worth tackling both issues at once.

**Comment by kandayin** [comment]
> open -a Neovide --args -- {filename}

this works! Would you happen to know how to properly pass this as an env var? eg

```bash
export EDITOR="open -a Neovide --args -- " <-- this still gives 2 tabs when $EDITOR is invoked
```

**Comment by falcucci** [comment]
Using `open` as the application launcher, both ways already mentioned by @fredizzimo should work.

The only thing to notice is that on macOS it will always launch taking `/` as the root path for those specific cases, so I would update the suggestion adding the `$PWD` for neovide context:

```bash
open -a Neovide --args $PWD/<file> --no-tabs
```

For who wants the duplicated tab issue fixed, the workaround would be disabling neovide `tabs` to make it appears as Buffer only using
```bash
--no-tabs
```

or even at our [config.toml](https://neovide.dev/config-file.html#available-settings) file.

```toml
tabs = false
```

**Comment by chrisgrieser** [comment]
Unfortunately, `--args` only solves the problem for the commandline. When you open a file via the macOS Finder, it also does not set the arglist.

**Comment by fredizzimo** [comment]
Yes, it looks like we really have to wait until all the open requests are handled at startup and then pass all of them to Neovim. That it turn requires some hacks as described in https://github.com/neovide/neovide/issues/2609#issuecomment-2148227966

I'm re-opening this and the other issue, as I think it should be possible to implement both without workarounds for the users.

**Comment by fboundp** [comment]
I don’t think this will be possible, because what is happening with the open command is that the files are not being passed as arguments to the application. Rather, they trigger open file events to be sent after the application has started. When you use the —args option, that really does pass the arguments directly to Neovide.

**Comment by fredizzimo** [comment]
I think it is, but it needs a bit re arranging of the code. We should be to gather all the request to the open file events and delay the start of Neovim until `applicationDidFinishLaunching` is called. 

It's not that easy though, but it should be possible. In fact it might be worth also thinking about the multiwindow support at the same time https://github.com/neovide/neovide/issues/1332. I believe that one will require us to be able to spawn multiple Neovim instances, one for each window, all within the same Neovide instance.

**Comment by fboundp** [comment]
That’s ambitious. I like it :-)

I would love to see multi-window support— it is the only thing now I miss from MacVIM.

**Comment by brglng** [comment]
I am using [alpha-nvim](https://github.com/goolord/alpha-nvim), and I want it not to show (and modify the open buffer content) when Neovide is launched via "Open With" from Finder. I cannot find a way to distinguish if Neovide is currently being launched via "Open With" or not, as `vim.fn.argc()` always returns 0. Is there any workaround for this?
